src/CourseModule/service/course.service.ts

```diff
@@ -19,14 +19,18 @@ export class CourseService {
   @Transactional()
   public async add(newCourse: NewCourseDTO, file): Promise<Course> {
     const course = this.mapper.toEntity(newCourse);
-    const user: User = await this.userService.findById(newCourse.authorId);
-    course.author = user;
-
-
-    // eslint-disable-next-line require-atomic-updates
+    course.author = await this.userService.findById(newCourse.authorId);
     course.photoName = file.filename;
-
-    return this.repository.save(course);
+    let savedCourse: Course;
+    try {
+      savedCourse = await this.repository.save(course);
```

Comment:
```suggestion
      return await this.repository.save(course);
```
Pode retornar direto, mas deixa o await para cair no catch de falhar

Reply:
Eu tinha tentado com o return await, mas não tava funcionando. Vou tentar de novo

Reply:
Ajustado conforme requisitei